src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -273,13 +297,13 @@ void ProbeDirectory(string manifestDirectory, string featureBand)
             //  Assume that if there are any versioned subfolders, they are higher manifest versions than a workload manifest directly in the specified folder, if it exists
             if (manifestVersionDirectories.Any())
             {
-                return (manifestId, manifestVersionDirectories.First().directory);
+                return (manifestId, manifestVersionDirectories.First().directory, manifestVersionDirectories.First().version);
             }
             else if (File.Exists(Path.Combine(manifestDirectory, "WorkloadManifest.json")))
             {
-                return (manifestId, manifestDirectory);
+                return (manifestId, manifestDirectory, null);
```

Comment:
I think returning null for the version here will cause problems when we try to compute the hash for the workload set version.  So we probably need to actually read the manifest to get the version.

Reply:
> I think returning null for the version here will cause problems when we try to compute the hash for the workload set version. So we probably need to actually read the manifest to get the version.

This turned out to be a disaster when it came to tests. Tests are making invalid WorkloadManifest.json files, and when I try to parse them for a version, it crashes.

So the obvious good solution to that is to fix the tests, right? I should change the tests so they write valid WorkloadManifest.json files.

Except there are several problems. First, that's a horrible mess just from a "format a string in C#" perspective. That's made worse by the fact that this is in a number of tests, so I can't correct it in one place and then have everything else build off of that—I need to correct both the place that creates the WorkloadManifest.json file and the place that assumes it's present and correct. Second, the previous format assumed certain parameters would be present in the WorkloadManifest.json file—and that's no longer true. The format has changed, which means I can't include the same information that was previously there. Third, it occurred to me that that problem might be true going forward as well; that is, future developers may decide to change the WorkloadManifest.json format, in which case they'd have to change all these tests all over again.

I wasted several hours on this today. My eventual resolution was to throw up my hands and just put in a try/catch statement instead. I think that should work and should make the tests happy. It's not exactly ideal, since I think we should notice if there's a problem and throw an error, but I think it's a fairly good solution as far as solutions go, and much easier from an implementation perspective than what I was trying to do.

Let me know if you disagree or see a better path forward.

Reply:
Can you add a comment explaining that the catch is there for tests that don't write valid manifests?

I'm not sure what you mean about the manifest format changing.  I don't believe it has changed significantly since it shipped.